### PR TITLE
fix(encoder): always fire LV_EVENT_LONG_PRESSED to indev callback

### LINKED ISSUE #5991: v9.1.0 - Encoder does not fire LV_EVENT_LONG_PRESSED
### LVGL version

v9.1.0

### What happened?

I have an encoder indev. It successfully rotates and handles short clicks, but long clicks does not issue LV_EVENT_LONG_PRESSED, although they do issue LV_EVENT_LONG_PRESSED_REPEATED regularly

### How to reproduce?

```c
lv_indev_add_event_cb(indev_encoder, long_press_cb, LV_EVENT_LONG_PRESSED, NULL); // never fires
lv_indev_add_event_cb(indev_encoder, long_press_cb, LV_EVENT_LONG_PRESSED_REPEAT, NULL); // fires regularly when long pressing
lv_indev_add_event_cb(indev_encoder, long_press_cb, LV_EVENT_PRESSED, NULL); // fires on any press

```

### PR DESCRIPTION
**Pull request by aeroniemi**
### Description of the feature or fix
Fixes #5991 

Currently LV_EVENT_LONG_PRESSED isn't fired by encoder buttons if the current obj is editable or scrollable - this fixes that by ensuring that it runs every time irrespective of the obj state.

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) version [v3.4.12](https://github.com/szepeviktor/astyle/releases/tag/v3.4.12) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

**Review by liamHowatt — COMMENTED** [review]
Looks good, thanks for creating this!

I'd like to get @kisvegabor to verify whether this one event is okay or whether all of the press events should get this treatment too, in one PR. That way we can point to this commit as "the commit where all press events are emitted for encoder indevs" instead of the transition being spread across many PRs.

Otherwise looks good.

**Comment by kisvegabor** [comment]
Thank you @aeroniemi.

I'm not sure if we need to support all the events here. Adding support for long press makes sens as it's related to the edit/navigation mode change and the user might be curious about it. 

The other events are omitted intentionally and it might be confusing to receive all the events even if they don't do anything. (E.g. `LV_EVENT_SHORT_CLICKED` on entering edit mode)

The only other indev event that I'd add unconditionally is `RELEASE` because it's very basic. 

It feel a little bit arbitrary and I'm not sure this is the right approach. So I'd be curious to hear your opinion.

**Comment by liamHowatt** [comment]
Aha, I see. Long press does make the most sense because the repeat event is always emitted and it just makes sense for long press to also be emitted. I see what you mean about the other events not having as strong of a case to be added, especially short click as you mentioned.

I guess my train of thought was that a user might want their encoder button to always have its events reflect the physical encoder button regardless of the state of the UI. But if that were the case, then they would write their own logic outside LVGL if it was that critical.

I could see release being added but I'd ask if press should also be added. Currently press is only emitted if it causes "editing" mode to switch on.

**TLDR** I think it's fine to leave this PR as-is.

**Review by liamHowatt — APPROVED** [review]

**Comment by kisvegabor** [comment]
> TLDR I think it's fine to leave this PR as-is.

Okay, let"s leave it as it is :slightly_smiling_face:

**Review by kisvegabor — APPROVED** [review]

**Review by FASTSHIFT — APPROVED** [review]
